benchmarks/maxtext_xpk_runner.py

```diff
@@ -357,15 +456,14 @@ def _get_pathways_specific_flags(wl_config: WorkloadConfig):
       else ''
   )
 
-  proxy_flags = pw_config.proxy_flags + wl_config.model.xla_flags
+  proxy_flags = _get_pathways_proxy_flags(wl_config)
 
   pathways_specific_flags = (
       f' {server_image_flag} '
       f' {proxy_server_image_flag} '
       f' {remote_python_sidecar_image_flag} '
       f' --termination-grace-period-seconds=300 '
       f' --pathways-gcs-location={wl_config.base_output_directory} '
-      f' --restart-on-user-code-failure'
```

Comment:
I think this was needed in XPK, so that Maxtext code failures leads to restart of the whole JobSet. Are we removing it by mistake?

Reply:
No this is intentional actually - by default XPK has switched to a different approach in the `develop` branch. This is the flag that's been replaced - https://github.com/AI-Hypercomputer/xpk/blob/develop/src/xpk/parser/workload.py#L599-L611.

The implementation of this already includes exit codes of 1 (amongst others) - https://github.com/AI-Hypercomputer/xpk/blob/develop/src/xpk/commands/workload.py#L733-L747